src/Featurizers/StandardScaleWrapperFeaturizer.h

```diff
@@ -0,0 +1,331 @@
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    StandardScaleWrapperEstimatorImpl(AnnotationMapsPtr pAllColumnAnnotations, size_t colIndex, bool withMean, bool withStd);
+    ~StandardScaleWrapperEstimatorImpl(void) override = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(StandardScaleWrapperEstimatorImpl);
+
+private:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Data
+    // |
+    // ----------------------------------------------------------------------
+    size_t const                            _colIndex;
+    // TODO: change bools to bit flags
+    bool   const                            _withMean;
+    bool   const                            _withStd;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Methods
+    // |
+    // ----------------------------------------------------------------------
+    bool begin_training_impl(void) override;
+    FitResult fit_impl(typename BaseType::InputType const *, size_t) override;
+    void complete_training_impl(void) override;
+
+    // MSVC has problems when the definition is separate from the declaration
+    typename BaseType::TransformerUniquePtr create_transformer_impl(void) override {
+        // ----------------------------------------------------------------------
+        using StandardStatisticalAnnotationData = Components::StandardStatisticalAnnotationData<InputT>;
+        using StatisticalMetricsEstimator       = Components::StatisticalMetricsEstimator<InputT, MaxNumTrainingItemsV>;
+
+        using StandardDeviationAnnotationData   = Components::StandardDeviationAnnotationData;
+        using StandardDeviationEstimator        = Components::StandardDeviationEstimator<InputT, MaxNumTrainingItemsV>;
+        // ----------------------------------------------------------------------
+        // if _withMean is false, average is set to 0
+        std::double_t average = 0;
+        // if _withStd is false, deviation is set to 1
+        std::double_t deviation = 1;
+        if (_withMean) {
+            StandardStatisticalAnnotationData const &        average_stats(StatisticalMetricsEstimator::get_annotation_data(this->get_column_annotations(), _colIndex, Components::StatisticalMetricsEstimatorName));
+            average = average_stats.Average;
+        }
+        if (_withStd) {
+            StandardDeviationAnnotationData   const &        deviation_stats(StandardDeviationEstimator::get_annotation_data(this->get_column_annotations(), _colIndex, Components::StandardDeviationEstimatorName));
+            deviation = static_cast<std::double_t>(deviation_stats.StandardDeviation);
+        }
+
+        // check if deviation is 0
+        // if it's 0, change it to 1
+        if (deviation == 0.0) {
```

Comment:
why does the deviation need to be 1? 0 is a valid standard deviation.

Reply:
0 is a valid standard deviation indeed but in this featurizer, standard deviation need to be devided to get unit standard deviation and 0 is not a valid denominator.